### PR TITLE
Add support for Inelukis MP3 Patch

### PR DESCRIPTION
**Pull request by Ghabry**
Differences to the real patch because we use the BGM Api directly:
 - It uses our BGM Api directly, this means compared to the real patch it will not overaly with the normal BGM which must be stopped manually (Needs testing in games if this is a problem)
 - The Patch allows playing multiple MP3s at the same time, but who needs this. (and Disharmony fixed this problem)
 - No support for the "loop" statement on line 2, it will always loop.
 - Our code does not care about the format, it can play all formats we support (but nobody uses this obviously).

This patch is enabled for all games, I see no risk in doing this. In the unlikely case a game really uses a filename with ".link" and then fails, we will see this in an Android bug report ;)

Algorithm:
When .link found it reads the first line and converts that path into a canonical one using the game directory as a root and passes it to FindDefault which finds the file. Simple ^^

### CODE REVIEW

**Comment by carstene1ns** [comment]
One thing I do not like about this is the generated noise in the log. You will get 3 lines related to one file in some cases:
```
[2016-08-02 23:10:09] Debug: Ineluki link file: ../RPG_Maker Games/CalmFalls/Music/17-Prisonic_Fairytale.link.wav
[2016-08-02 23:10:09] Debug: Linking to: ..\mp3\17-Prisonic_Fairytale.MP3
[2016-08-02 23:10:09] Debug: MP3 heuristic: ../RPG_Maker Games/CalmFalls/mp3/17-Prisonic_Fairytale.MP3
```
Can we get rid of the heuristic line already? It should be well tested enough until we get another bug reports after dropping sdl_mixer... :P Also it would be okay to join the Ineluki lines in one.

**Comment by Ghabry** [comment]
Done

File: src/game_system.cpp [hunk 1]
```diff
@@ -307,23 +308,54 @@ void Game_System::SetTransition(int which, int transition) {
...
+	Audio().BGM_Play(path, data.current_music.volume, data.current_music.tempo, data.current_music.fadein);
 }
 
 void Game_System::OnSeReady(FileRequestResult* result, int volume, int tempo) {
 	auto item = se_request_ids.find(result->file);
 	if (item != se_request_ids.end()) {
 		se_request_ids.erase(item);
 	}
-
+
```
Comment (carstene1ns) [inline src/game_system.cpp]: :scream:

**Comment by carstene1ns** [comment]
If this works for games using the escape_symbol for directory traversal, then looks good to me.

**Comment by Ghabry** [comment]
Now that you say this, I forgot something. The read line must be converted from game encoding to utf8. >.>

**Comment by carstene1ns** [comment]
Works for me correctly with Calm Falls & Nebulus intro. Merging.

**Pull request merged**